### ISSUE TITLE
Query timeout error: "Query timed out", 30

### ISSUE THREAD

**Issue by benthamite** [org-roam/org-roam#2206]
### Description
I recently migrated to a new laptop. Unfortunately, I am unable to get org-roam to work on this new machine. Upon attempting to synchronize the database, an error message results, and no db file is created.

#### Steps to Reproduce

1. Load Emacs
2. Run `org-roam-db-sync`
3. [Emacs freezes for 30 seconds]
4. The following message is displayed in the echo area: 'Query timeout error: "Query timed out", 30'

#### Backtrace

```
Debugger entered--Lisp error: (emacsql-timeout "Query timed out" 30)
  signal(emacsql-timeout ("Query timed out" 30))
  #f(compiled-function (connection &optional timeout) "Block until CONNECTION is waiting for further input." #<bytecode 0xdcd203a8e03c0ff>)(#<emacsql-sqlite-connection emacsql-sqlite-connection-41406e88>)
  apply(#f(compiled-function (connection &optional timeout) "Block until CONNECTION is waiting for further input." #<bytecode 0xdcd203a8e03c0ff>) #<emacsql-sqlite-connection emacsql-sqlite-connection-41406e88> nil)
  emacsql-wait(#<emacsql-sqlite-connection emacsql-sqlite-connection-41406e88>)
  #f(compiled-function (connection sql &rest args) "Send SQL s-expression to CONNECTION and return the results." #<bytecode -0x1de985a4cf2deafb>)(#<emacsql-sqlite-connection emacsql-sqlite-connection-41406e88> [:pragma (= busy-timeout $s1)] 15000)
  apply(#f(compiled-function (connection sql &rest args) "Send SQL s-expression to CONNECTION and return the results." #<bytecode -0x1de985a4cf2deafb>) #<emacsql-sqlite-connection emacsql-sqlite-connection-41406e88> ([:pragma (= busy-timeout $s1)] 15000))
  emacsql(#<emacsql-sqlite-connection emacsql-sqlite-connection-41406e88> [:pragma (= busy-timeout $s1)] 15000)
  #f(compiled-function (#<emacsql-sqlite-connection emacsql-sqlite-connection-41406e88> (:file "/Users/pablostafforini/.emacs.d/var/org/org-roam.d..."))
  apply(#f(compiled-function
  #f(compiled-function (&rest args) #<bytecode 0x24e89310fe37eb8>)(#<emacsql-sqlite-connection emacsql-sqlite-connection-41406e88> (:file "/Users/pablostafforini/.emacs.d/var/org/org-roam.d..."))
  apply(#f(compiled-function (&rest args) #<bytecode 0x24e89310fe37eb8>) #<emacsql-sqlite-connection emacsql-sqlite-connection-41406e88> (:file "/Users/pablostafforini/.emacs.d/var/org/org-roam.d..."))
  initialize-instance(#<emacsql-sqlite-connection emacsql-sqlite-connection-41406e88> (:file "/Users/pablostafforini/.emacs.d/var/org/org-roam.d..."))
  #f(compiled-function (class &rest slots) "Default constructor for CLASS `eieio-default-superclass'.\nSLOTS are the initialization slots used by `initialize-instance'.\nThis static method is called when an object is constructed.\nIt allocates the vector used to represent an EIEIO object, and then\ncalls `initialize-instance' on that object." #<bytecode 0x151baf8e0318ed83>)(emacsql-sqlite-connection :file "/Users/pablostafforini/.emacs.d/var/org/org-roam.d...")
  apply(#f(compiled-function (class &rest slots) "Default constructor for CLASS `eieio-default-superclass'.\nSLOTS are the initialization slots used by `initialize-instance'.\nThis static method is called when an object is constructed.\nIt allocates the vector used to represent an EIEIO object, and then\ncalls `initialize-instance' on that object." #<bytecode 0x151baf8e0318ed83>) emacsql-sqlite-connection (:file "/Users/pablostafforini/.emacs.d/var/org/org-roam.d..."))
  make-instance(emacsql-sqlite-connection :file "/Users/pablostafforini/.emacs.d/var/org/org-roam.d...")
  emacsql-sqlite("/Users/pablostafforini/.emacs.d/var/org/org-roam.d...")
  org-roam-db()
  org-roam-db-sync()
  eval((org-roam-db-sync) t)
  eval-expression((org-roam-db-sync) nil nil 127)
  funcall-interactively(eval-expression (org-roam-db-sync) nil nil 127)
  call-interactively(eval-expression nil nil)
  command-execute(eval-expression)

```

#### Expected Results

The org-roam database synchronizes.

#### Actual Results

The org-roam database does not synchronize.

### Environment

- Emacs: GNU Emacs 28.1 (build 1, aarch64-apple-darwin21.5.0, Carbon Version 165 AppKit 2113.5)
 of 2022-05-25
- Framework: N/A
- Org: Org mode version 9.5 (9.5-g6c8f8f @ /Users/pablostafforini/.emacs.d/straight/build/org/)
- Org-roam: v2.1.0-23-g0ecbd3a

**Comment by nobiot** [comment]
Does your new Mac have C compiler installed? I suggest you ensure this and then re-install Org-roam. Your issue is probably the missing executable for emacsql-sqlite (or the executable from the old machine is for a different CPU architecture). If you have a compatible executable, check the permission.

**Comment by jethrokuan** [comment]
On your emacs version it might also be possible to use built-in sqlite support: like

```
(setq org-roam-database-connector 'sqlite-builtin)
```

**Comment by benthamite** [comment]
Thank you for the quick replies!

@nobiot, I do seem to have a C compiler installed:

``` shell
$ which clang 
/usr/bin/clang
$ which gcc
/usr/bin/gcc
```
When you say, "Your issue is probably the missing executable for emacsql-sqlite", is there anything I can do to test whether this is the case? I'm sorry, I'm not a programmer.

@jethrokuan

I had to upgrade to the latest org-roam version, since for some reason Straight had downloaded a version (v2.1.0-23-g0ecbd3a) prior to the inclusion of the `org-roam-database-connector` user option. However, I get the backtrace below. Thoughts on what may be going on?

``` emacs-lisp
(use-package emacsql-sqlite-builtin)
(use-package org-roam
:custom
	(org-roam-database-connector 'sqlite-builtin)
	(org-roam-directory ps/dir-org-roam))
```

### Backtrace
```
Debugger entered--Lisp error: (file-missing "Cannot open load file" "No such file or directory" "sqlite")
  require(sqlite)
  eval-buffer(#<buffer  *load*> nil "/Users/pablostafforini/.emacs.d/straight/build/ema..." nil t)  ; Reading at buffer position 1227
  load-with-code-conversion("/Users/pablostafforini/.emacs.d/straight/build/ema..." "/Users/pablostafforini/.emacs.d/straight/build/ema..." nil t)
  require(emacsql-sqlite-builtin)
  org-roam-db--conn-fn()
  org-roam-db()
  org-roam-db-sync(nil)
  funcall-interactively(org-roam-db-sync nil)
  call-interactively(org-roam-db-sync record nil)
  command-execute(org-roam-db-sync record)
  helm-M-x-execute-command(org-roam-db-sync)
  helm-execute-selection-action-1()
  helm-execute-selection-action()
  helm-internal((((name . "Emacs Commands history") (init #f(compiled-function () #<bytecode 0x1b4a4a74f825c1af>)) (candidates . helm-candidates-in-buffer) (keymap keymap (keymap ...) keymap (21 . helm-M-x-universal-argument) keymap (127 . helm-mode-delete-char-backward-maybe) (27 keymap ...) (C-return . helm-cr-empty-string) keymap (8388640 closure ... nil ... ...) (8388719 closure ... nil ... ...) (8388713 closure ... nil ... ...) (8388725 closure ... nil ... ...) (8388716 closure ... nil ... ...) (8388715 closure ... nil ... ...) (8388714 closure ... nil ... ...) (8388654 closure ... nil ... ...) (8388652 closure ... nil ... ...) (8388717 closure ... nil ... ...) (26 . helm-select-action) (8388709 . ps/helm-ff-run-open-file-externally) (f13 lambda nil ... ...) (f12 lambda nil ... ...) (f11 lambda nil ... ...) (f10 lambda nil ... ...) (f9 lambda nil ... ...) (f8 lambda nil ... ...) (f7 lambda nil ... ...) (f6 lambda nil ... ...) (f5 lambda nil ... ...) (f4 lambda nil ... ...) ...) (action . helm-type-command-actions) (persistent-action . helm-M-x-persistent-action) (persistent-help . "Describe this command") (help-message . helm-M-x-help-message) (requires-pattern . 0) (filtered-candidate-transformer helm-M-x-transformer-no-sort #f(compiled-function (candidates source) #<bytecode -0x1293558f10b9e90b>) helm-fuzzy-highlight-matches) (volatile . t) (match identity) (fuzzy-match . t) (redisplay . identity) (nomark . t) (coerce . helm-symbolify) (header-line . #("C-j: Describe this command (keep..." 0 3 ...)) (multimatch . t) (must-match . t) (group . helm-command) (data . #f(compiled-function () #<bytecode -0xcff404ddcec0817>)) (get-line . buffer-substring-no-properties) (search helm-mm-exact-search helm-mm-search helm-candidates-in-buffer-search-default-fn helm-fuzzy-search)) ((name . "Emacs Commands") (init #f(compiled-function () #<bytecode -0x1d15842ab03c4716>)) (candidates . helm-candidates-in-buffer) (keymap keymap (keymap ...) keymap (21 . helm-M-x-universal-argument) keymap (127 . helm-mode-delete-char-backward-maybe) (27 keymap ...) (C-return . helm-cr-empty-string) keymap (8388640 closure ... nil ... ...) (8388719 closure ... nil ... ...) (8388713 closure ... nil ... ...) (8388725 closure ... nil ... ...) (8388716 closure ... nil ... ...) (8388715 closure ... nil ... ...) (8388714 closure ... nil ... ...) (8388654 closure ... nil ... ...) (8388652 closure ... nil ... ...) (8388717 closure ... nil ... ...) (26 . helm-select-action) (8388709 . ps/helm-ff-run-open-file-externally) (f13 lambda nil ... ...) (f12 lambda nil ... ...) (f11 lambda nil ... ...) (f10 lambda nil ... ...) (f9 lambda nil ... ...) (f8 lambda nil ... ...) (f7 lambda nil ... ...) (f6 lambda nil ... ...) (f5 lambda nil ... ...) (f4 lambda nil ... ...) ...) (action . helm-type-command-actions) (persistent-action . helm-M-x-persistent-action) (persistent-help . "Describe this command") (help-message . helm-M-x-help-message) (requires-pattern . 0) (filtered-candidate-transformer helm-M-x-transformer-no-sort #f(compiled-function (candidates source) #<bytecode -0x1293558f10b9e90b>) helm-fuzzy-highlight-matches) (volatile . t) (match identity) (fuzzy-match . t) (redisplay . identity) (nomark . t) (coerce . helm-symbolify) (header-line . #("C-j: Describe this command (keep..." 0 3 ...)) (multimatch . t) (must-match . t) (group . helm-command) (data . #f(compiled-function () #<bytecode 0x2ce1f0850025a39>)) (get-line . buffer-substring-no-properties) (search helm-mm-exact-search helm-mm-search helm-candidates-in-buffer-search-default-fn helm-fuzzy-search))) nil "M-x " nil nil "*helm M-x*" nil nil helm-M-x-input-history)
  apply(helm-internal ((((name . "Emacs Commands history") (init #f(compiled-function () #<bytecode 0x1b4a4a74f825c1af>)) (candidates . helm-candidates-in-buffer) (keymap keymap (keymap ...) keymap (21 . helm-M-x-universal-argument) keymap (127 . helm-mode-delete-char-backward-maybe) (27 keymap ...) (C-return . helm-cr-empty-string) keymap (8388640 closure ... nil ... ...) (8388719 closure ... nil ... ...) (8388713 closure ... nil ... ...) (8388725 closure ... nil ... ...) (8388716 closure ... nil ... ...) (8388715 closure ... nil ... ...) (8388714 closure ... nil ... ...) (8388654 closure ... nil ... ...) (8388652 closure ... nil ... ...) (8388717 closure ... nil ... ...) (26 . helm-select-action) (8388709 . ps/helm-ff-run-open-file-externally) (f13 lambda nil ... ...) (f12 lambda nil ... ...) (f11 lambda nil ... ...) (f10 lambda nil ... ...) (f9 lambda nil ... ...) (f8 lambda nil ... ...) (f7 lambda nil ... ...) (f6 lambda nil ... ...) (f5 lambda nil ... ...) (f4 lambda nil ... ...) (f3 lambda nil ... ...) (f2 lambda nil ... ...) (menu-bar keymap ...) (help keymap ...) (23 . #f(compiled-function () ... #<bytecode -0x188e3707a240e7b2>)) (f1 lambda nil ... ...) ...) (action . helm-type-command-actions) (persistent-action . helm-M-x-persistent-action) (persistent-help . "Describe this command") (help-message . helm-M-x-help-message) (requires-pattern . 0) (filtered-candidate-transformer helm-M-x-transformer-no-sort #f(compiled-function (candidates source) #<bytecode -0x1293558f10b9e90b>) helm-fuzzy-highlight-matches) (volatile . t) (match identity) (fuzzy-match . t) (redisplay . identity) (nomark . t) (coerce . helm-symbolify) (header-line . #("C-j: Describe this command (keeping se..." 0 3 ...)) (multimatch . t) (must-match . t) (group . helm-command) (data . #f(compiled-function () #<bytecode -0xcff404ddcec0817>)) (get-line . buffer-substring-no-properties) (search helm-mm-exact-search helm-mm-search helm-candidates-in-buffer-search-default-fn helm-fuzzy-search)) ((name . "Emacs Commands") (init #f(compiled-function () #<bytecode -0x1d15842ab03c4716>)) (candidates . helm-candidates-in-buffer) (keymap keymap (keymap ...) keymap (21 . helm-M-x-universal-argument) keymap (127 . helm-mode-delete-char-backward-maybe) (27 keymap ...) (C-return . helm-cr-empty-string) keymap (8388640 closure ... nil ... ...) (8388719 closure ... nil ... ...) (8388713 closure ... nil ... ...) (8388725 closure ... nil ... ...) (8388716 closure ... nil ... ...) (8388715 closure ... nil ... ...) (8388714 closure ... nil ... ...) (8388654 closure ... nil ... ...) (8388652 closure ... nil ... ...) (8388717 closure ... nil ... ...) (26 . helm-select-action) (8388709 . ps/helm-ff-run-open-file-externally) (f13 lambda nil ... ...) (f12 lambda nil ... ...) (f11 lambda nil ... ...) (f10 lambda nil ... ...) (f9 lambda nil ... ...) (f8 lambda nil ... ...) (f7 lambda nil ... ...) (f6 lambda nil ... ...) (f5 lambda nil ... ...) (f4 lambda nil ... ...) (f3 lambda nil ... ...) (f2 lambda nil ... ...) (menu-bar keymap ...) (help keymap ...) (23 . #f(compiled-function () ... #<bytecode -0x188e3707a240e7b2>)) (f1 lambda nil ... ...) ...) (action . helm-type-command-actions) (persistent-action . helm-M-x-persistent-action) (persistent-help . "Describe this command") (help-message . helm-M-x-help-message) (requires-pattern . 0) (filtered-candidate-transformer helm-M-x-transformer-no-sort #f(compiled-function (candidates source) #<bytecode -0x1293558f10b9e90b>) helm-fuzzy-highlight-matches) (volatile . t) (match identity) (fuzzy-match . t) (redisplay . identity) (nomark . t) (coerce . helm-symbolify) (header-line . #("C-j: Describe this command (keeping se..." 0 3 ...)) (multimatch . t) (must-match . t) (group . helm-command) (data . #f(compiled-function () #<bytecode 0x2ce1f0850025a39>)) (get-line . buffer-substring-no-properties) (search helm-mm-exact-search helm-mm-search helm-candidates-in-buffer-search-default-fn helm-fuzzy-search))) nil "M-x " nil nil "*helm M-x*" nil nil helm-M-x-input-history))
  helm((((name . "Emacs Commands history") (init #f(compiled-function () #<bytecode 0x1b4a4a74f825c1af>)) (candidates . helm-candidates-in-buffer) (keymap keymap (keymap ...) keymap (21 . helm-M-x-universal-argument) keymap (127 . helm-mode-delete-char-backward-maybe) (27 keymap ...) (C-return . helm-cr-empty-string) keymap (8388640 closure ... nil ... ...) (8388719 closure ... nil ... ...) (8388713 closure ... nil ... ...) (8388725 closure ... nil ... ...) (8388716 closure ... nil ... ...) (8388715 closure ... nil ... ...) (8388714 closure ... nil ... ...) (8388654 closure ... nil ... ...) (8388652 closure ... nil ... ...) (8388717 closure ... nil ... ...) (26 . helm-select-action) (8388709 . ps/helm-ff-run-open-file-externally) (f13 lambda nil ... ...) (f12 lambda nil ... ...) (f11 lambda nil ... ...) (f10 lambda nil ... ...) (f9 lambda nil ... ...) (f8 lambda nil ... ...) (f7 lambda nil ... ...) (f6 lambda nil ... ...) (f5 lambda nil ... ...) (f4 lambda nil ... ...) ...) (action . helm-type-command-actions) (persistent-action . helm-M-x-persistent-action) (persistent-help . "Describe this command") (help-message . helm-M-x-help-message) (requires-pattern . 0) (filtered-candidate-transformer helm-M-x-transformer-no-sort #f(compiled-function (candidates source) #<bytecode -0x1293558f10b9e90b>) helm-fuzzy-highlight-matches) (volatile . t) (match identity) (fuzzy-match . t) (redisplay . identity) (nomark . t) (coerce . helm-symbolify) (header-line . #("C-j: Describe this command (keep..." 0 3 ...)) (multimatch . t) (must-match . t) (group . helm-command) (data . #f(compiled-function () #<bytecode -0xcff404ddcec0817>)) (get-line . buffer-substring-no-properties) (search helm-mm-exact-search helm-mm-search helm-candidates-in-buffer-search-default-fn helm-fuzzy-search)) ((name . "Emacs Commands") (init #f(compiled-function () #<bytecode -0x1d15842ab03c4716>)) (candidates . helm-candidates-in-buffer) (keymap keymap (keymap ...) keymap (21 . helm-M-x-universal-argument) keymap (127 . helm-mode-delete-char-backward-maybe) (27 keymap ...) (C-return . helm-cr-empty-string) keymap (8388640 closure ... nil ... ...) (8388719 closure ... nil ... ...) (8388713 closure ... nil ... ...) (8388725 closure ... nil ... ...) (8388716 closure ... nil ... ...) (8388715 closure ... nil ... ...) (8388714 closure ... nil ... ...) (8388654 closure ... nil ... ...) (8388652 closure ... nil ... ...) (8388717 closure ... nil ... ...) (26 . helm-select-action) (8388709 . ps/helm-ff-run-open-file-externally) (f13 lambda nil ... ...) (f12 lambda nil ... ...) (f11 lambda nil ... ...) (f10 lambda nil ... ...) (f9 lambda nil ... ...) (f8 lambda nil ... ...) (f7 lambda nil ... ...) (f6 lambda nil ... ...) (f5 lambda nil ... ...) (f4 lambda nil ... ...) ...) (action . helm-type-command-actions) (persistent-action . helm-M-x-persistent-action) (persistent-help . "Describe this command") (help-message . helm-M-x-help-message) (requires-pattern . 0) (filtered-candidate-transformer helm-M-x-transformer-no-sort #f(compiled-function (candidates source) #<bytecode -0x1293558f10b9e90b>) helm-fuzzy-highlight-matches) (volatile . t) (match identity) (fuzzy-match . t) (redisplay . identity) (nomark . t) (coerce . helm-symbolify) (header-line . #("C-j: Describe this command (keep..." 0 3 ...)) (multimatch . t) (must-match . t) (group . helm-command) (data . #f(compiled-function () #<bytecode 0x2ce1f0850025a39>)) (get-line . buffer-substring-no-properties) (search helm-mm-exact-search helm-mm-search helm-candidates-in-buffer-search-default-fn helm-fuzzy-search))) nil "M-x " nil nil "*helm M-x*" nil nil helm-M-x-input-history)
  apply(helm ((((name . "Emacs Commands history") (init #f(compiled-function () #<bytecode 0x1b4a4a74f825c1af>)) (candidates . helm-candidates-in-buffer) (keymap keymap (keymap ...) keymap (21 . helm-M-x-universal-argument) keymap (127 . helm-mode-delete-char-backward-maybe) (27 keymap ...) (C-return . helm-cr-empty-string) keymap (8388640 closure ... nil ... ...) (8388719 closure ... nil ... ...) (8388713 closure ... nil ... ...) (8388725 closure ... nil ... ...) (8388716 closure ... nil ... ...) (8388715 closure ... nil ... ...) (8388714 closure ... nil ... ...) (8388654 closure ... nil ... ...) (8388652 closure ... nil ... ...) (8388717 closure ... nil ... ...) (26 . helm-select-action) (8388709 . ps/helm-ff-run-open-file-externally) (f13 lambda nil ... ...) (f12 lambda nil ... ...) (f11 lambda nil ... ...) (f10 lambda nil ... ...) (f9 lambda nil ... ...) (f8 lambda nil ... ...) (f7 lambda nil ... ...) (f6 lambda nil ... ...) (f5 lambda nil ... ...) (f4 lambda nil ... ...) (f3 lambda nil ... ...) (f2 lambda nil ... ...) (menu-bar keymap ...) (help keymap ...) (23 . #f(compiled-function () ... #<bytecode -0x188e3707a240e7b2>)) (f1 lambda nil ... ...) ...) (action . helm-type-command-actions) (persistent-action . helm-M-x-persistent-action) (persistent-help . "Describe this command") (help-message . helm-M-x-help-message) (requires-pattern . 0) (filtered-candidate-transformer helm-M-x-transformer-no-sort #f(compiled-function (candidates source) #<bytecode -0x1293558f10b9e90b>) helm-fuzzy-highlight-matches) (volatile . t) (match identity) (fuzzy-match . t) (redisplay . identity) (nomark . t) (coerce . helm-symbolify) (header-line . #("C-j: Describe this command (keeping se..." 0 3 ...)) (multimatch . t) (must-match . t) (group . helm-command) (data . #f(compiled-function () #<bytecode -0xcff404ddcec0817>)) (get-line . buffer-substring-no-properties) (search helm-mm-exact-search helm-mm-search helm-candidates-in-buffer-search-default-fn helm-fuzzy-search)) ((name . "Emacs Commands") (init #f(compiled-function () #<bytecode -0x1d15842ab03c4716>)) (candidates . helm-candidates-in-buffer) (keymap keymap (keymap ...) keymap (21 . helm-M-x-universal-argument) keymap (127 . helm-mode-delete-char-backward-maybe) (27 keymap ...) (C-return . helm-cr-empty-string) keymap (8388640 closure ... nil ... ...) (8388719 closure ... nil ... ...) (8388713 closure ... nil ... ...) (8388725 closure ... nil ... ...) (8388716 closure ... nil ... ...) (8388715 closure ... nil ... ...) (8388714 closure ... nil ... ...) (8388654 closure ... nil ... ...) (8388652 closure ... nil ... ...) (8388717 closure ... nil ... ...) (26 . helm-select-action) (8388709 . ps/helm-ff-run-open-file-externally) (f13 lambda nil ... ...) (f12 lambda nil ... ...) (f11 lambda nil ... ...) (f10 lambda nil ... ...) (f9 lambda nil ... ...) (f8 lambda nil ... ...) (f7 lambda nil ... ...) (f6 lambda nil ... ...) (f5 lambda nil ... ...) (f4 lambda nil ... ...) (f3 lambda nil ... ...) (f2 lambda nil ... ...) (menu-bar keymap ...) (help keymap ...) (23 . #f(compiled-function () ... #<bytecode -0x188e3707a240e7b2>)) (f1 lambda nil ... ...) ...) (action . helm-type-command-actions) (persistent-action . helm-M-x-persistent-action) (persistent-help . "Describe this command") (help-message . helm-M-x-help-message) (requires-pattern . 0) (filtered-candidate-transformer helm-M-x-transformer-no-sort #f(compiled-function (candidates source) #<bytecode -0x1293558f10b9e90b>) helm-fuzzy-highlight-matches) (volatile . t) (match identity) (fuzzy-match . t) (redisplay . identity) (nomark . t) (coerce . helm-symbolify) (header-line . #("C-j: Describe this command (keeping se..." 0 3 ...)) (multimatch . t) (must-match . t) (group . helm-command) (data . #f(compiled-function () #<bytecode 0x2ce1f0850025a39>)) (get-line . buffer-substring-no-properties) (search helm-mm-exact-search helm-mm-search helm-candidates-in-buffer-search-default-fn helm-fuzzy-search))) nil "M-x " nil nil "*helm M-x*" nil nil helm-M-x-input-history))
  helm(:sources (((name . "Emacs Commands history") (init #f(compiled-function () #<bytecode 0x1b4a4a74f825c1af>)) (candidates . helm-candidates-in-buffer) (keymap keymap (keymap ...) keymap (21 . helm-M-x-universal-argument) keymap (127 . helm-mode-delete-char-backward-maybe) (27 keymap ...) (C-return . helm-cr-empty-string) keymap (8388640 closure ... nil ... ...) (8388719 closure ... nil ... ...) (8388713 closure ... nil ... ...) (8388725 closure ... nil ... ...) (8388716 closure ... nil ... ...) (8388715 closure ... nil ... ...) (8388714 closure ... nil ... ...) (8388654 closure ... nil ... ...) (8388652 closure ... nil ... ...) (8388717 closure ... nil ... ...) (26 . helm-select-action) (8388709 . ps/helm-ff-run-open-file-externally) (f13 lambda nil ... ...) (f12 lambda nil ... ...) (f11 lambda nil ... ...) (f10 lambda nil ... ...) (f9 lambda nil ... ...) (f8 lambda nil ... ...) (f7 lambda nil ... ...) (f6 lambda nil ... ...) (f5 lambda nil ... ...) (f4 lambda nil ... ...) ...) (action . helm-type-command-actions) (persistent-action . helm-M-x-persistent-action) (persistent-help . "Describe this command") (help-message . helm-M-x-help-message) (requires-pattern . 0) (filtered-candidate-transformer helm-M-x-transformer-no-sort #f(compiled-function (candidates source) #<bytecode -0x1293558f10b9e90b>) helm-fuzzy-highlight-matches) (volatile . t) (match identity) (fuzzy-match . t) (redisplay . identity) (nomark . t) (coerce . helm-symbolify) (header-line . #("C-j: Describe this command (keep..." 0 3 ...)) (multimatch . t) (must-match . t) (group . helm-command) (data . #f(compiled-function () #<bytecode -0xcff404ddcec0817>)) (get-line . buffer-substring-no-properties) (search helm-mm-exact-search helm-mm-search helm-candidates-in-buffer-search-default-fn helm-fuzzy-search)) ((name . "Emacs Commands") (init #f(compiled-function () #<bytecode -0x1d15842ab03c4716>)) (candidates . helm-candidates-in-buffer) (keymap keymap (keymap ...) keymap (21 . helm-M-x-universal-argument) keymap (127 . helm-mode-delete-char-backward-maybe) (27 keymap ...) (C-return . helm-cr-empty-string) keymap (8388640 closure ... nil ... ...) (8388719 closure ... nil ... ...) (8388713 closure ... nil ... ...) (8388725 closure ... nil ... ...) (8388716 closure ... nil ... ...) (8388715 closure ... nil ... ...) (8388714 closure ... nil ... ...) (8388654 closure ... nil ... ...) (8388652 closure ... nil ... ...) (8388717 closure ... nil ... ...) (26 . helm-select-action) (8388709 . ps/helm-ff-run-open-file-externally) (f13 lambda nil ... ...) (f12 lambda nil ... ...) (f11 lambda nil ... ...) (f10 lambda nil ... ...) (f9 lambda nil ... ...) (f8 lambda nil ... ...) (f7 lambda nil ... ...) (f6 lambda nil ... ...) (f5 lambda nil ... ...) (f4 lambda nil ... ...) ...) (action . helm-type-command-actions) (persistent-action . helm-M-x-persistent-action) (persistent-help . "Describe this command") (help-message . helm-M-x-help-message) (requires-pattern . 0) (filtered-candidate-transformer helm-M-x-transformer-no-sort #f(compiled-function (candidates source) #<bytecode -0x1293558f10b9e90b>) helm-fuzzy-highlight-matches) (volatile . t) (match identity) (fuzzy-match . t) (redisplay . identity) (nomark . t) (coerce . helm-symbolify) (header-line . #("C-j: Describe this command (keep..." 0 3 ...)) (multimatch . t) (must-match . t) (group . helm-command) (data . #f(compiled-function () #<bytecode 0x2ce1f0850025a39>)) (get-line . buffer-substring-no-properties) (search helm-mm-exact-search helm-mm-search helm-candidates-in-buffer-search-default-fn helm-fuzzy-search))) :prompt "M-x " :buffer "*helm M-x*" :history helm-M-x-input-history)
  helm-M-x-read-extended-command([org-capture-goto-last-stored table--measure-max-width log-edit-remember-comment calendar-forward-week telega-msg-forward-marked-or-at-point-to-multiple-chats tramp-completion-handle-file-name-all-completions Set\ Slice el-name speedbar-check-vc-this-line mediawiki-prompt-for-page ido-prev-match magit-revision-filter-files-on-follow helm-completing-read-inbuffer-default-handler yas-x-prompt gnus-article-nndoc-name &error rng-parse-validate-file magit-xref-insert-button speedbar-file-regexp gnus-agent-group-covered-p orcp-collapse-numeric-range edebug-mode-on-hook org-habit-scheduled reftex-untie-multifile-symbols autoload-find-file nnoo-map-functions traversed compilation-error-list :where orb-plist :odt-styles-file vc-annotate :mac mediawiki-site-domain helm-highlight-bookmark rng-match-infer-start-tag-namespace bibtex-completion-apa-format-authors-abbrev traversing\ a\ directory gnus-summary-limit-to-marks bibtex-completion-candidates nnmail-message-id-cache-file helm-grep-save-results-1 filename-and-process smime-buffer-as-string-region org-get-org-file Futura core-accessing-symbol transient:magit-log:--dense bibtex-text-in-string ! ...])
  helm-M-x(nil)
  funcall-interactively(helm-M-x nil)
  call-interactively(helm-M-x nil nil)
  command-execute(helm-M-x)
```

**Comment by nobiot** [comment]
> is there anything I can do to test whether this is the case?

It's easy. Check if you have a file named `emacsql-sqlite` in your Straight directory where it installs packages. It should be somewhere under `emacsql-sqlite` directory. I use the built-in `package.el` instead of Straight and it puts the file under a sub-directory named `sqlite`. In my case, the path is: 

```
/home/nobiot/.config/emacs/elpa/emacsql-sqlite-20220218.1543/sqlite
```

If you find the file, check if it has the "executable" permission for you -- I will assume you understand what I'm talking about. There was a case where someone restored backup into a new machine (I guess you have done this for your mac machine). It was done by a super user, not your usual user name. They didn't have the permission to execute the file -- similar error (I can't remember if it was the exact same error, though).

For the `(org-roam-database-connector 'sqlite-builtin)` point, there are two things:

1. It's the Emacs version that matters first of all. You're using 28.1 so I don't believe it is relevant. The Emacs built-in SQLite is still experimental and currently only part of development branch for Emacs 29.x -- not released; you'd have to compile Emacs from source to get it.

2. It's weird Straight downloaded a rather old commit of Org-roam v2.1.0-23-g0ecbd3a. The latest version is v.2.2.2 since the end of April. You might like to check if your Straight is working properly -- I don't use it so I cannot be of help here. But I often see people deleting the package repository (Org-roam) and re-install it (at least in the context of Doom Emacs). 

Just on the side as a suggestion, and you can completely disregard this. I'm no developer; I am under the impression that Straight is more effective if you are a developer and use Git extensively? It might be of less headache for you if you stick to the built-in `package.el` -- just a gentle suggestion. I'm very happy with it and have no headache managing integrity and versions of packages. Please disregard my mumbling if you are happy with Straight.

**Comment by benthamite** [comment]
Hi @nobiot 

Thank you again for your help. (Your general willingness to help org-roam users is admirable!)

I checked and the executable is there, and I have permissions to run it.

A user reports the same problem [here](https://github.com/skeeto/emacsql/issues/96), and another person says it was due to a misconfigured PATH (solved by enabling `exec-path-from-shell`). I'm launching Emacs from the terminal, though, so it's not exactly that. But I wonder whether it's related to that, since I can't get mu4e to find mbsync and pass, as reported [here](https://github.com/djcb/mu/issues/2264#issuecomment-1140022951) (note I don't think it's a bug but I can't remove the "bug" tag).

Unless you or others have further hypotheses about what may be going on, I think I will follow your suggestion and drop Straight. I thought it would simplify package management, but its behavior is sometimes erratic, or at least that's how it feels to me, a mere novice.

**Comment by jethrokuan** [comment]
> I had to upgrade to the latest org-roam version, since for some reason Straight had downloaded a version (v2.1.0-23-g0ecbd3a) prior to the inclusion of the org-roam-database-connector user option. However, I get the backtrace below. Thoughts on what may be going on?

ah your Emacs version is still not recent enough :)

**Comment by nobiot** [comment]
What happens if you delete emacsql-sqlite file from Finder (or Terminal) and then launch Emacs and Org-roam? Will it recompile the executable correctly? 

Also I thought there was a variable you can check that sets the path to the executable—even if you are running Emacs in terminal. I may be mistaken but I’d need to check later. Going to bed now

**Comment by benthamite** [comment]
Okay, I have now decisively established that the problem isn't specific to org-roam, but is a general problem with the environmental variables in my system. I've concluded this after experiencing various other packages, which I hadn't used until now since migrating to my new machine, also failing to find various other programs, like curl, ledger, etc.

What puzzles me is that this happens even when I run Emacs from the command line, and even when I set the PATH to the location of the relevant program. E.g. I run `setenv` to set `PATH` to `/usr/bin`, but in an Emacs shell `which sqlite3` fails to find sqlite3, although it is in `/usr/bin/sqlite3`. (By contrast, running `which sqlite3` in Eshell does return `/usr/bin/sqlite3`.)

Anyway, although my issue isn't resolved, I'll close this now, since it is not specifically related to org-roam. Thank you both for you help!
